### PR TITLE
webos: Limit maximum feedahead time

### PR DESCRIPTION
**Pull request by sundermann**
## Description

Limit the maximum amount of seconds the pipeline can be fed ahead.

## Motivation and context

Some users still reported excessive buffering after the bufferinfo changes. This enforces a hard limit of 1s maximum feedahead data. This is also implemented similarily in the chromium webOS sourcecode: https://github.com/webosose/chromium84-temp/blob/df04aa361ad493d47003a80ebfa3034dfc9de6e0/src/media/neva/webos/media_platform_api_webos_smp.cc#L513-L517C56

## How has this been tested?

OLED77C28LB (webOS 7/22)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp [hunk 1]
```diff
@@ -346,6 +350,7 @@ void CDVDVideoCodecStarfish::Reset()
 
   CLog::LogF(LOGDEBUG, "CDVDVideoCodecStarfish: Reset");
   m_starfishMediaAPI->flush();
+  m_fedTimestamp = -1ns;
```
Comment (wsnipex) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp:353]: would be nice to use a constant for "invalid / not enough data"

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ constexpr unsigned int MIN_BUFFER_LEVEL = 0;
 constexpr unsigned int MAX_BUFFER_LEVEL = 0;
 constexpr unsigned int MIN_SRC_BUFFER_LEVEL = 1 * 1024 * 1024; // 1 MB
 constexpr unsigned int MAX_SRC_BUFFER_LEVEL = 8 * 1024 * 1024; // 8 MB
+constexpr std::chrono::nanoseconds MAX_FEED_AHEAD_TIME = 1s;
```
[the remark this answers is not in the document]
Comment (sundermann) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp:42]: I didn't notice any downside. If so we can always increase it later or make it configurable

**Comment by wsnipex** [comment]
this seems to break DV, I only get audio with a black screen. Today's nightly is ok

**Comment by sundermann** [comment]
Added some more logic to make sure pipeline start is not limited by this change

**Review by wsnipex — APPROVED** [review]
seems to improve video startup time, especially for non DV files

**Comment by craigcarnell** [comment]
Tested today, improved start up times as @wsnipex mentioned

**Comment by wsnipex** [comment]
I'm not yet sure if this is the culprit, but I've had several picture freezes watching a DV movie yesterday.

**Comment by craigcarnell** [comment]
@wsnipex @sundermann  Any progress on this PR? I've been using three seconds for feedahead time since moving to a NAS with gigabit ethernet and it seems to work fairly well.

**Comment by sundermann** [comment]
I've monitored my network traffic and I cannot see any excessive buffering anymore. It's possible that the previous behavior has been fixed as a sideeffect of one of the recent changes in the decoder. Closing the PR for now

**Pull request closed without merging**